repo/common/common.go

```diff
@@ -2,26 +2,50 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 )
 
+// Find dynamic map key names passed  as Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\[([^\[\]]*)\]`)
+
+// Normalization for supporting arbitrary dynamic keys with dots:
+// Gateway.PublicGateways["gw.example.com"].UseSubdomains
+// Pinning.RemoteServices["pins.example.org"].Policies.MFS.Enable
+func keyToLookupData(key string) (normalizedKey string, dynamicKeys map[string]string) {
+	bracketedKeys := bracketsRe.FindAllString(key, -1)
+	dynamicKeys = make(map[string]string, len(bracketedKeys))
+	normalizedKey = key
+	for i, mapKeySegment := range bracketedKeys {
+		mapKey := strings.TrimLeft(mapKeySegment, "[\"")
+		mapKey = strings.TrimRight(mapKey, "\"]")
+		placeholder := fmt.Sprintf("mapKey%d", i)
```

Comment:
This could be problematic if someone uses `mapKey0` as a name of one of their fields. If we're going to use a placeholder approach (as opposed to unwinding the string segment by segment approach) then maybe there is a reserved/invalid JSON character we can use where we first validate that the string doesn't have any of those characters.

Reply:
I switched placeholders to UUIDs to avoid conflicts with real keys while keeping magic to minimum  (732adebaf)
(uuid lib was already a transitional dependency, so should not impact binary size).